.github/workflows/bash.yml

```diff
@@ -63,8 +63,8 @@ jobs:
 
       - name: Install Cylc dependencies in the container
         run: |
-          docker exec bash python3.7 -m pip install six==1.12
-          docker exec -w /root/cylc-flow bash python3.7 -m pip install .[all]
+          docker exec bash python3.12 -m pip install six==1.12
+          docker exec -w /root/cylc-flow bash python3.12 -m pip install .[all]
```

Comment:
> OCI runtime exec failed: exec failed: unable to start container process: exec: "python3.12": executable file not found in $PATH: unknown
>
> @oliver-sanders any ideas?

Python 3.12 isn't present in the container. The container appears to be external to the repo:

https://github.com/cylc/cylc-docker/pkgs/container/cylc-bash-testing

---

I'm not a big fan of this testing approach as it is more flexible to `conda install bash=<version>` than to `docker pull bask:<version>` because we then have control over the base image.

Have suggested conda'ising this solution, but never got around to it. Also it's tricky making sure that the Conda installed Bash is always used.